File: sorokac_u2/uvTosd.m
function [s, d] = uvTosd(u, v, uk, vk)
    %Convert u, v to oblique aspect
    
    % Compute coordinate differences
    dv = vk - v;
    
    %Latitude
    sarg = sin(u) * sin(uk) + cos(u) * cos(uk) .* cos(dv);
    s = asin(sarg);
    
    %Longitude   
    num = -(sin(dv) .* cos(u));
    denom = cos(u) .* sin(uk) .* cos(dv) - sin(u) .* cos(uk);
    
    d = atan2(num, denom);
end

File: uloha_1/uvtosd.py
from math import *

def uvTosd(u, v, uk, vk):
    #Convert u, v to the oblique aspect
    dv = vk - v

    # Latitude
    sarg = sin(u) * sin(uk) + cos(u) * cos(dv)*cos(uk)
    s = asin(sarg)
    
    #Longitude
    num = sin(dv) * cos(u)
    denom = -sin(u) * cos(uk) + cos(u) * cos(dv) * sin(uk)

    d = -atan2(num, denom)

    return s, d